### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by kevin-10**

### CODE REVIEW

File: Scripts/Services/Dungeons/TheExodusEncounter/Mobiles/ClockworkExodus.cs [hunk 1]
```diff
@@ -92,20 +92,24 @@ public static void DistributeRandomArtifact(BaseCreature bc, Type[] typelist)
...
+            {
+                Container pack = to.Backpack;
 
-            if (pack == null || !pack.TryDropItem(to, artifact, false))
-                to.BankBox.DropItem(artifact);
+                if (pack == null || !pack.TryDropItem(to, artifact, false))
+                    to.BankBox.DropItem(artifact);
 
-            to.SendLocalizedMessage(502088); // A special gift has been placed in your backpack.
+                to.SendLocalizedMessage(502088); // A special gift has been placed in your backpack.
```
Comment (djredman99) [inline Scripts/Services/Dungeons/TheExodusEncounter/Mobiles/ClockworkExodus.cs:107]: Is there not a separate message for dropping in bankbox (when pack is full) ?

File: Scripts/Mobiles/Bosses/Navery/Navrey.cs [hunk 2]
```diff
@@ -77,7 +77,7 @@ public static void DistributeRandomArtifact(BaseCreature bc, Type[] typelist)
         {
             int random = Utility.Random(typelist.Length);
             Item item = Loot.Construct(typelist[random]);
-            DistributeArtifact(DemonKnight.FindRandomPlayer(bc), item);
+            DistributeArtifact(bc.RandomPlayerWithLootingRights(), item);
```
Comment (djredman99) [inline Scripts/Mobiles/Bosses/Navery/Navrey.cs:80]: Much better!

File: Scripts/Services/Dungeons/CovetousVoidSpawn/Region.cs [hunk 3]
```diff
@@ -68,7 +68,7 @@ public override void OnDeath(Mobile m)
 
                     if (chance > Utility.RandomDouble())
                     {
-                        Mobile mob = DemonKnight.FindRandomPlayer((BaseCreature)m);
+                        Mobile mob = ((BaseCreature)m).RandomPlayerWithLootingRights();
```
Comment (djredman99) [inline Scripts/Services/Dungeons/CovetousVoidSpawn/Region.cs:71]: Possible InvalidCastException?

File: Scripts/Services/Expansions/High Seas/Spawners/BountyQuestSpawner.cs [hunk 4]
```diff
@@ -664,19 +664,18 @@ public static void FillHold(BaseGalleon galleon)
...
-                    if (0.25 > Utility.RandomDouble())
+                    hold.DropItem(new MaritimeCargo());
+                    hold.DropItem(new MaritimeCargo());
+
+                    if (galleon is OrcishGalleon)
                     {
                         hold.DropItem(new MaritimeCargo());
 
-                        if (0.1 > Utility.RandomDouble())
+                        if (Utility.RandomBool())
```
Comment (djredman99) [inline Scripts/Services/Expansions/High Seas/Spawners/BountyQuestSpawner.cs:674]: Should there be a chance for a 4th cargo?  (3 or 4 from Pirate ships from [Heritage Bug](https://trueuo.com/threads/plunderbeacon-merchant-ship-cargo-not-dropping.2230/)

**Review by djredman99 — COMMENTED** [review]
I have some inline comments :P

File: Scripts/Services/Dungeons/CovetousVoidSpawn/Region.cs (hunk 3 above)
Comment (kevin-10) [reply Scripts/Services/Dungeons/CovetousVoidSpawn/Region.cs:71]: ill check, that was already cast with the previous code I repleaced.

Comment (kevin-10) [reply Scripts/Services/Dungeons/CovetousVoidSpawn/Region.cs:71]: Yes:
if (m is CovetousCreature && ((CovetousCreature)m).VoidSpawn)

File: Scripts/Services/Expansions/High Seas/Spawners/BountyQuestSpawner.cs (hunk 4 above)
Comment (kevin-10) [reply Scripts/Services/Expansions/High Seas/Spawners/BountyQuestSpawner.cs:674]: There is 4. The first 2 are for all ship types. Then, orcish galleons (pirates) drop another, plus a chance for a fourth.

if (Server.Engines.Points.PointsSystem.RisingTide.Enabled)
                {
                    hold.DropItem(new MaritimeCargo()); 1
                    hold.DropItem(new MaritimeCargo()); 2

                    if (galleon is OrcishGalleon)
                    {
                        hold.DropItem(new MaritimeCargo()); 3

                        if (Utility.RandomBool())
                        {
                            hold.DropItem(new MaritimeCargo()); 4
                        }
                    }
                }

File: Scripts/Services/Dungeons/TheExodusEncounter/Mobiles/ClockworkExodus.cs (hunk 1 above)
Comment (kevin-10) [reply Scripts/Services/Dungeons/TheExodusEncounter/Mobiles/ClockworkExodus.cs:107]: There is no cliloc that supports that.

Comment (djredman99) [reply Scripts/Services/Dungeons/TheExodusEncounter/Mobiles/ClockworkExodus.cs:107]: What about
1111852 -- You have received a special gift!
1079730 -- The item has been placed into your bank box.
OR
1156322 -- A reward of ~1_ITEM~ has been placed in your bank.

Comment (kevin-10) [reply Scripts/Services/Dungeons/TheExodusEncounter/Mobiles/ClockworkExodus.cs:107]: I have no idea how EA handles it. There are a ton of special/artifact drops with different messages, most of which go to our bank if your pack is full. I know doom artifact drops, there is no special message when this happens.

Comment (TrueUO) [reply Scripts/Services/Dungeons/TheExodusEncounter/Mobiles/ClockworkExodus.cs:107]: Hard to test but this makes sense - 1156322

Comment (djredman99) [reply Scripts/Services/Dungeons/TheExodusEncounter/Mobiles/ClockworkExodus.cs:107]: Lots of dumb people out there, we have to help them understand what happened to their item :)

Comment (djredman99) [reply Scripts/Services/Dungeons/TheExodusEncounter/Mobiles/ClockworkExodus.cs:107]: I like the combination of the first two, personally.

**Review by TrueUO — APPROVED** [review]
Going to approve this and get these fixes in. I fully agree with Redman, something I'll go back and add, even if its a custom its a minor QOL.bump.

**Pull request merged**